cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudSQLMySQLAction.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.cloudsql.mysql;
+
+import com.google.common.collect.ImmutableMap;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.etl.api.action.Action;
+import io.cdap.plugin.db.batch.action.AbstractDBAction;
+import io.cdap.plugin.db.batch.action.QueryConfig;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Action that runs a MySQL command on a CloudSQL instance database.
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name(CloudSQLMySQLConstants.PLUGIN_NAME)
+@Description("Action that runs a MySQL command on a CloudSQL instance database")
+public class CloudSQLMySQLAction extends AbstractDBAction {
+
+  private final CloudSQLMySQLActionConfig cloudsqlMysqlActionConfig;
+
+  public CloudSQLMySQLAction(CloudSQLMySQLActionConfig cloudsqlMysqlActionConfig) {
+    super(cloudsqlMysqlActionConfig, false);
+    this.cloudsqlMysqlActionConfig = cloudsqlMysqlActionConfig;
+  }
+
```

Comment:
one other thing that I think you should add is validation for the connection name. Override the configurePIpeline method in each plugin like below, wher eyou call ``super.configurePipeline()``, and then also additionally verify that the connection name is in the right format (3 strings separated by ``:``?)

Reply:
Added a `checkConnectionName()` method in a new `CloudSQLMySQLUtil` class for the verification.